internal/pkg/aws/ec2/ec2.go

```diff
@@ -186,23 +161,65 @@ func (c *EC2) HasDNSSupport(vpcID string) (bool, error) {
 	return aws.BoolValue(resp.EnableDnsSupport.Value), nil
 }
 
-// ListVPCSubnets lists all subnets given a VPC ID.
-func (c *EC2) ListVPCSubnets(vpcID string, opts ...ListVPCSubnetsOpts) ([]string, error) {
-	respSubnets, err := c.subnets(Filter{
+// VPCSubnets are all subnets within a VPC.
+type VPCSubnets struct {
+	Public  []VPCResource
+	Private []VPCResource
+}
+
+// ListVPCSubnets lists all subnets given a VPC ID. Note that public subnets
+// are subnets associated with a route table that routed by an internet gateway.
```

Comment:
the wording of my fix is kinda weird but maybe a little clearer? 
```suggestion
// are subnets associated with an internet gateway through a route table.
```